src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/AttributeSection.test.tsx

```diff
@@ -193,6 +219,77 @@ describe("AttributeSection", () => {
     const dateInput2 = await screen.findByTestId("CalendarIcon");
     expect(dateInput).toBeInTheDocument();
     expect(dateInput2).toBeInTheDocument();
+
+    fireEvent.change(dateInput, { target: { value: "01/01/2023" } });
+    expect(dateInput.value).toBe("01/01/2023");
+
+    const addButton = screen.getByTestId("AddCircleOutlineIcon");
+    expect(addButton).toBeInTheDocument();
+    userEvent.click(addButton);
+  });
+
+  it("shows RatioInput on selecting the Ratio type", async () => {
+    const assessmentElement: AssessmentPerformed = new AssessmentPerformed();
+    const { container } = render(
+      <AttributeSection selectedDataElement={assessmentElement} />
+    );
+
+    const attributeSelectBtn = screen.getByRole("button", {
+      name: "Attribute Select Attribute",
+    });
+    expect(attributeSelectBtn).toBeInTheDocument();
+
+    userEvent.click(attributeSelectBtn);
+
+    const attributeSelect = await screen.findByRole("listbox");
+    const attributeOptions = within(attributeSelect).getAllByRole("option");
+    expect(attributeOptions).toHaveLength(7);
+
+    userEvent.click(within(attributeSelect).getByText(/result/i));
+    const attributeInput = within(attributeSelectBtn.parentElement).getByRole(
+      "textbox",
+      { hidden: true }
+    );
+    expect(attributeInput).toBeInTheDocument();
+    expect(attributeInput).toHaveValue("Result");
+    const typeSelectBtn = await screen.findByRole("button", {
+      name: /type/i,
+    });
+    expect(typeSelectBtn).toBeInTheDocument();
+    userEvent.click(typeSelectBtn);
+    const typeSelect = await screen.findByRole("listbox");
+    expect(typeSelect).toBeInTheDocument();
+    const typeOptions = within(typeSelect).getAllByRole("option");
+    expect(typeOptions.length).toEqual(9);
+    fireEvent.click(within(typeSelect).getByText("Ratio"));
+
+    await waitFor(() => {
```

Comment:
do we need such a large waitFor? Or can we leverage `await screen.findByRole` in a better way? (And note that findByXYZ also accepts a timeout option if needed

Reply:
I tried different timeout numbers but couldn't get the tests pass.